src/platform/forms-system/src/js/utilities/createStringifyFormReplacer.js

```diff
@@ -0,0 +1,81 @@
+import omit from 'platform/utilities/data/omit';
+import { replaceEscapedCharacters } from 'platform/forms-system/src/js/utilities/replaceEscapedCharacters';
+
+/**
+ *
+ * @param {Object} object
+ * @returns {boolean}
+ */
+const isObjectEmpty = object => {
+  const fields = Object.keys(object);
+
+  return (
+    fields.length === 0 || fields.every(field => object[field] === undefined)
+  );
+};
+
+function isPartialAddress(data) {
+  return (
+    data &&
+    typeof data.country !== 'undefined' &&
+    (!data.street || !data.city || (!data.postalCode && !data.zipcode))
+  );
+}
+
+/**
+ * Replaces objects (including arrays) with other values
+ *
+ * @param {Object} object
+ * @param {ReplacerOptions} [options] - Options for the replacer
+ * @returns {*} The replacement value
+ */
+const replaceObject = (object, options) => {
+  // Clean up empty objects in arrays
+  if (Array.isArray(object)) {
+    const newValues = object.filter(v => !isObjectEmpty(v));
```

Comment:
Can you double check this part? I think during array iteration, we need to do the full check again, not just an object check, for example if we have an address in an array.

Reply:
Good catch. I'm not sure the original code supported nested arrays due to the use of `filter` instead of `reduce`, but I've added a spec to make sure they are supported moving forward.